### ISSUE TITLE
TextInput.hint_text too long for TextInput.size_hint_x locks the app

### ISSUE THREAD

**Issue by LogicalDash** [kivy/kivy#4244]
If I have a TextInput whose `size_hint_x` makes it too small to hold all its `hint_text`, my app freezes (apparently in an infinite loop) instead of truncating the `hint_text` as I'd expect.

Steps to reproduce:

1. Make a TextInput, give it long hint_text and small size_hint_x (or put it in a BoxLayout so that a "large" size_hint_x still makes it too small)
2. Launch app

Expected behavior: get an app window that I can interact with
Actual behavior: app eats 100% CPU and doesn't display anything

Example code:
```
from kivy.base import runTouchApp
from kivy.lang import Builder
from kivy.uix.boxlayout import BoxLayout


class TestBoxLayout(BoxLayout):
    pass


Builder.load_string("""
<TestBoxLayout>:
    orientation: 'vertical'
    Widget:
        id: foo
    Widget:
        id: bar
    BoxLayout:
        TextInput:
            hint_text: 'Some hint text'
            size_hint_x: 0.5
        Widget:
            id: spam
        Widget:
            id: eggs
""")


if __name__ == '__main__':
    runTouchApp(TestBoxLayout())
```

**Comment by dessant** [comment]
It works as expected with 1.9.1.

**Comment by dessant** [comment]
Introduced by https://github.com/kivy/kivy/pull/4186.
@kived ping.

**Comment by dessant** [comment]
@kived has found the buggy commit: 29e7f851a8ae58ba018c9a0a3d639b8a20c0b69e.

**Comment by LogicalDash** [comment]
This is happening again, but I made (I think) a more general fix in #4445
